app/views/shared/_table.html.erb

```diff
@@ -28,8 +26,19 @@
    $('#data-table').DataTable({
      "order": [[ 1, "desc" ]],
      "stateSave": true,
+     "stateSaveParams": function (settings, data) {
+       const savableKeys = ["search"]
+       Object.keys(data).forEach(function(key) {
+         if (!savableKeys.includes(key)) {
+           delete data[key]
+         }
+       })
+     },
      "columnDefs": [
-       { "targets": "hidden", "visible": false },
+       {
+         "targets": <%= hidden_headers.map { |hh| headers.index(hh) } %>,
```

Comment:
This seems flaky, if there is an mismatch between lists, then "nil" will be output, which doesn't mean anything in JS.

How about `<%= JSON.generate(hidden_headers.map { |hh| headers.index(hh) }) %>`

Reply:
That makes sense.  I didn't consider that possibility.

Reply:
Fair point. Will fix.